src/huggingface_hub/repository.py

```diff
@@ -869,14 +869,119 @@ def git_checkout(self, revision, create_branch_ok=False):
                 except subprocess.CalledProcessError as exc:
                     raise EnvironmentError(exc.stderr)
 
-    def push_to_hub(self, commit_message: str = "commit files to HF hub") -> str:
+    def tag_exists(self, tag_name: str) -> bool:
+        """
+        Check if a tag exists or not
+        """
+        try:
+            git_tags = subprocess.run(
+                ["git", "tag"],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        git_tags = git_tags.split("\n")
+        return tag_name in git_tags
+
+    def delete_tag(self, tag_name: str) -> bool:
```

Comment:
Having a `remote` argument here that would default to `"origin"` would be welcome!

Reply:
Can do